### ISSUE TITLE
[Bug Report] Progress does not save + reader page does not change

### ISSUE THREAD

**Issue by DJScias** [hkalexling/Mango#105]
**Describe the bug**
1. Progress doesn't save or change. Every time I stop reading my progress remains stuck at the value it was before v0.12.2.

2. Not sure if related, but when I scrolled down the reader the page number would change from `/1` to `/2` etc... in the url. Now it just stays stuck at `/1`.

**To Reproduce**
1. Open a manga (new or in progress, doesn't matter)
2. Scroll down
3. Refresh or go to main url/stop reading

Refreshed:
4. You're back on page 1

Main url/stop reading:
4. Your reading progress hasn't changed, even if you chose "next entry".

**Expected behavior**
Refreshed:
Go back to the last page you were on (or whatever it was before).

Main url/stop reading:
Progress is correctly updated.

**Environment (please complete the following information):**
 - OS: Synology DS918+, DSM 6.2.3-25426 
 - Browser: N/A
 - Mango Version: v0.12.1

**Docker (if you are running Mango in a Docker container)**
```
sudo docker create \
  --name=mango \
  -e PUID=1024 \
  -e PGID=100 \
  -p 9000:9000/tcp \
  -v /volume1/docker/Mango/config:/root/.config/mango \
  -v /volume4/Plex4/Mango:/root/mango \
  --restart always \
	hkalexling/mango:v0.12.2
```

**Additional context**
This only started to be an issue after v0.12.2, the last version everything worked fine for me (with "next entry" as well) was v0.11.0

Hopefully this info helps, if you need more please do let me know.

**Comment by hkalexling** [comment]
Hmm, I couldn't reproduce it. Do you see any error messages in the browser console?

**Comment by DJScias** [comment]
None and I keep consistently reproducing it.

Just tried "watch from beginning" a chapter, then chose "next entry" and the URL remains stuck at `/1` still. I then close the tab, go to a new one and my "last watching" wasn't ever updated, still stuck before I even chose "watch from beginning".

Anything else I can do? For v0.12.2 I just deleted the container and re-created it as new as usual.

**Comment by hkalexling** [comment]
This is likely a frontend issue. The JS code that's responsible for updating the URL and sending the reading progress to the backend API is not working correctly. 

What browser are you using? Could you try it in another browser or try deleting the browser cache?

**Comment by DJScias** [comment]
Hello there,

I mainly use Brave as a browser (worked perfectly since I started using it).

I've now tried it both in Chrome and Edge as well and each browser had something broken to varying degrees as well:

Chrome & Edge: The page indicator (`/1` and `/2`) in the URL does update but not always. However reading progress is not saved either.

How I tested:
1. Opened the manga and scrolled down to see if page indicator works in URL.
2. Close the manga at a random page (3, 4, 7) and go to main page to see if progress has updated, unfortunately still stuck at 0.0%.

If possible, I can set you up with my account on my mango server (it's running a reverse proxy so it's available everywhere) and then you can see it for yourself? I can easily switch the password so you have a temporary one.

I mean, anything to figure this out. Because switching browsers is a bit annoying (Brava is based on chromium like Chrome and (nowadays) Edge) but even on the other two my progress doesn't not save.

**Comment by hkalexling** [comment]
I still can't reproduce it on my end :(

Yes, it would be great if you could create a temporary account for me to test. You can email me the credentials at email@hkalexling.com. Thanks!

**Comment by hkalexling** [comment]
Thanks for your email and I think I have found the problem!

![test](https://user-images.githubusercontent.com/7845831/93612240-dbeee800-fa01-11ea-971d-b042f4be9668.png)

The "Tomb Raider King" manga you are reading has the webtoon-style long pages (see the screenshot above where your server info has been blurred out), so you have to scroll far enough to get to the second page. This is the reason why the URL doesn't seem to be updating.

Since v0.12.0, the web reader sends the reading progress to the backend every five pages, so the progress will only be saved when you've reached page 5, 10, and so on. Please try to scroll down further or just use the menu to jump to say page 8 and see if the progress is saved.

**Comment by hkalexling** [comment]
Saving every five pages works well for normal manga but I agree for webtoons it could break the user expectations. Let me think about it and see how we can improve.

**Comment by DJScias** [comment]
I'm glad to see there's some progress.

However in my browser it still doesn't update the page number on Brave even when I keep scrolling down.

Image before page 1 -> 2 transition (the big black bar)
![image](https://user-images.githubusercontent.com/439655/93613366-12603f00-f9d1-11ea-9b6f-242abb775772.png)

Image after that, it still says `/1` at the top bar.
![image](https://user-images.githubusercontent.com/439655/93613457-2e63e080-f9d1-11ea-8d82-fe20891b15b5.png)

I'm not sure how Brave could be affected by it or what is going wrong here, no errors anywhere though. :/

**Comment by hkalexling** [comment]
Hmm, that's weird. I just tested with Brave on MacOS and it works without any issue. 

A few questions:
- Can you confirm that both the URL and the progress saving work on your other browsers?
- In Brave, please clear the browser cache and storage and see if the problem persists. You can find the detailed instructions here https://github.com/hkalexling/Mango/issues/34#issuecomment-625712194
- In Brave, when you click on a page, does it open up the pop-up menu?
- In Brave, when you scroll from page one to the last page, does it save your reading progress (assuming the chapter you are reading has more than five pages)?

**Comment by DJScias** [comment]
Hello there,

I'm sorry I had you running in a wild goose chase, I've figured out the issue with Brave's thing.

It was some tampermonkey script interfering with the URL changing for some reason? Once I disabled it the `/1` and `/2` part immediately changed.

However I can consistently reproduce the other bug with "next entry".

When I'm on chapter 15 and go to "next entry" and then go back it shows this
![image](https://user-images.githubusercontent.com/439655/93619706-358eec80-f9d9-11ea-9e1f-91cd7ed8ea8e.png)

However, if I go to "next entry" and read up until page 5 it'll show this:
![image](https://user-images.githubusercontent.com/439655/93619773-56efd880-f9d9-11ea-97ae-7c6d22f97a63.png)

Now if I go into the `/book/` view, it shows me this:
![image](https://user-images.githubusercontent.com/439655/93619852-71c24d00-f9d9-11ea-8138-16ffb3de22c6.png)

Chapter 15 is the first chapter I started since the new reader.. For some reason it's not marking "next entry" as "this chapter is 100%, mark it as read" and it just keeps it at the 66.7%.

So that's basically sorta my #1 issue, but the part where the URL doesn't update is fixed and was just some extension.

Sorry and thanks in advance for taking the time to look into it.

**Comment by hkalexling** [comment]
Ah, you are right! The Javascript should tell the backend to mark a chapter as 100% when it's scrolled to the end. My bad!

I will fix this ASAP and come up with a way to better handle the long pages in webtoons. Thanks for the bug report 👍

**Comment by DJScias** [comment]
Not to worry, the script that blocked stuff was actually one of my own that "facilitated" long webtoon pages.

It basically adds a near "invisible" button in the top-left
![image](https://user-images.githubusercontent.com/439655/93621021-3e80bd80-f9db-11ea-89ea-dd6c4d60d7eb.png)

When clicked on it, it'd add a cookie with the book's id and then from that point on it would set the margins between the imgs to 0 (simply explained) for the books that I have pressed the button on.

But thanks for looking into it, take your time with it. It's an amazing app so far and making a quick and dirty "remove the margins" to work for my webtoon was easy enough.

**Comment by hkalexling** [comment]
This bug has been fixed in [v0.12.3](https://github.com/hkalexling/Mango/releases/tag/v0.12.3). When the average height over width ratio in the chapter is over 2, the web reader will treat it as a webtoon chapter and update the progress on every page.
